File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # using hasmap
        hash_map = {}
        
        for i, n in enumerate(nums):
            diff = target - n
            if diff in hash_map:
                return [hash_map[diff], i]
            hash_map[n] = i
        return []

File: 1-Two-Sum.go
func twoSum(nums []int, target int) []int {
    n := len(nums)

    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            if i != j && nums[i] + nums[j] == target {
                return []int{i, j}
            }
        }
    }

    return []int{}
}